app/browse/page.tsx

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import ContentsSlider from '@/components/browse/ContentsSlider';
+import HomeNavBar from '@/components/browse/HomeNavBar';
+import HomeTop from '@/components/browse/HomeTop';
+import { TContent } from '@/types';
+import { getMovies } from '@/utils/Api';
+
+interface BrowseProps {
+  searchParams: { [key: string]: string | undefined };
+}
+
+const Browse = async ({ searchParams }: BrowseProps) => {
+  const Top = await getMovies('top_rated');
+  const Popular = await getMovies('popular');
+  const Upcoming = await getMovies('upcoming');
+  const Nowplaying = await getMovies('now_playing');
+
+  const defaultData = {
+    rank: 1,
+    category: 'Top Rate',
+    posterPath: Top[0].poster_path,
+  };
+
+  return (
+    <div className="bg-black w-full min-h-full pb-12 relative">
+      <HomeNavBar />
+      <HomeTop
+        posterPath={searchParams.posterPath || defaultData.posterPath}
+        rank={
+          searchParams.rank
+            ? Number(searchParams.rank)
+            : searchParams.posterPath
+            ? null
+            : defaultData.rank
+        }
+        category={searchParams.category || defaultData.category}
+      />
+      <ContentsSlider
+        title="Preview"
+        isRanking={false}
+        isPreview={true}
+        contents={Nowplaying}
+      />
+      <ContentsSlider title="Nigeria Today" isRanking={true} contents={Top} />
+      <ContentsSlider title="Popular" isRanking={false} contents={Popular} />
+      <ContentsSlider title="Upcoming" isRanking={false} contents={Upcoming} />
```

Comment:
정보를 리스트 선언 후 map 함수로 렌더링 한다면 가독성과 추후 슬라이더 추가시에도 편리할 것 같다는 개인적인 의견 드립니다 ㅎㅎ

```suggestion

const sliderData=[
    {
      title: "Preview",
      movieData: await getMovies('now_playing'),
      isRanking: false,
      isPreview:true
}, 
   {
      title: "Nigeria Today",
      movieData: await getMovies('top_rated'),
      isRanking: true,
      isPreview:false
},
   {
      title: "Popular",
      movieData: await getMovies('popular'),
      isRanking: false,
      isPreview:false
},
   {
      title: "Upcoming",
      movieData: await getMovies('upcoming'),
      isRanking: false,
      isPreview:false
}
]

....

{sliderData.map((data,index)=>(
    <ContentsSlider
        key={data.title}
        title={data.title}
        isRanking={data.isRanking}
        isPreview={data.isPreview}
        contents={data.movieData}
      />
))}

Reply:
저도 배워갑니다~!